enterprise/backend/test/metabase_enterprise/semantic_search/core_test.clj

```diff
@@ -0,0 +1,60 @@
+(ns metabase-enterprise.semantic-search.core-test
+  (:require
+   [clojure.test :refer :all]
+   [metabase-enterprise.semantic-search.pgvector-api :as semantic.pgvector-api]
+   [metabase-enterprise.semantic-search.test-util :as semantic.tu]
+   [metabase.search.appdb.core :as appdb]
+   [metabase.search.core :as search] [metabase.search.engine :as search.engine]
+   [metabase.search.settings :as search.settings]
+   [metabase.test :as mt]
+   [metabase.test.fixtures :as fixtures]))
+
+(use-fixtures :once (compose-fixtures
+                     (fixtures/initialize :db)
+                     #'semantic.tu/once-fixture))
+
+(deftest appdb-available-with-semantic
+  (mt/with-premium-features #{:semantic-search}
+    (mt/with-temporary-setting-values [search.settings/search-engine "semantic"]
+      (with-open [_ (semantic.tu/open-temp-index!)]
+        (semantic.tu/cleanup-index-metadata! semantic.tu/db semantic.tu/mock-index-metadata)
+        (search/init-index! {:force-reset? false, :re-populate? false})
```

Comment:
Did you still want the `init-index!` here, or leftover from previous implementation?